litestar/response/base.py

```diff
@@ -24,28 +22,172 @@
         ResponseHeaders,
         Scope,
         Send,
+        Serializer,
         TypeEncodersMap,
     )
 
+__all__ = ("ASGIResponse", "Response")
+
 T = TypeVar("T")
 
 
+class ASGIResponse:
+    """A low-level ASGI response class."""
+
+    __slots__ = (
+        "background",
+        "body",
+        "content_length",
+        "encoded_headers",
+        "encoding",
+        "is_head_response",
+        "status_code",
+    )
+
+    def __init__(
+        self,
+        *,
+        background: BackgroundTask | BackgroundTasks | None = None,
+        body: bytes = b"",
+        content_length: int | None = None,
+        cookies: list[Cookie] | None = None,
+        encoded_headers: list[tuple[bytes, bytes]] | None = None,
+        encoding: str = "utf-8",
+        headers: dict[str, Any] | None = None,
+        is_head_response: bool = False,
+        media_type: str | None = None,
```

Comment:
I could still allow this to receive `MediaType` enum values - it would clean up usage a bit.